ibm_cloud_sdk_core/utils.py

```diff
@@ -218,21 +222,30 @@ def __read_from_vcap_services(service_name: str) -> dict:
         A set of service configuration key-value pairs.
     """
     vcap_services = getenv('VCAP_SERVICES')
+    if not vcap_services:
+        vcap_services_file = getenv('VCAP_SERVICES_FILE')
+        if os.path.exists(vcap_services_file):
+            with open(vcap_services_file) as vcap_file:
+                vcap_services = json_import.load(vcap_file)
     vcap_service_credentials = {}
     if vcap_services:
         services = json_import.loads(vcap_services)
-
         for key in services.keys():
-            if key == service_name:
-                vcap_service_credentials = services[key][0]['credentials']
-                if vcap_service_credentials is not None and isinstance(vcap_service_credentials, dict):
-                    if vcap_service_credentials.get('username') and vcap_service_credentials.get('password'): # cf
-                        vcap_service_credentials['AUTH_TYPE'] = 'basic'
-                        vcap_service_credentials['USERNAME'] = vcap_service_credentials.get('username')
-                        vcap_service_credentials['PASSWORD'] = vcap_service_credentials.get('password')
-                    elif vcap_service_credentials.get('apikey'):  # rc
-                        vcap_service_credentials['AUTH_TYPE'] = 'iam'
-                        vcap_service_credentials['APIKEY'] = vcap_service_credentials.get('apikey')
-                    else: # no other auth mechanism is supported
-                        vcap_service_credentials = {}
+            for i in range(len(services[key])):
+                if vcap_service_credentials and isinstance(vcap_service_credentials, dict):
+                    break
+                if services[key][i].get('name') == service_name:
+                    vcap_service_credentials = services[key][i].get('credentials')
+        if not vcap_service_credentials:
+            if service_name in services.keys():
+                vcap_service_credentials = services.get(service_name)[0].get('credentials')
+
+        if vcap_service_credentials and isinstance(vcap_service_credentials, dict):
+            if vcap_service_credentials.get('username') and vcap_service_credentials.get('password'): # cf
```

Comment:
I'm really confused by this... we seem to be returning `vcap_service_credentials` such that it will contain the "credentials" field of the service entry (as a dictionary... e.g. "username", "password", "apikey", "url", etc.) plus all the additional properties that we explicitly set in there (e.g. "USERNAME", "PASSWORD, "IAM_APIKEY", "URL", etc.).
I think we should return a dictionary that contains only the well-defined properties (uppercase names) that will be used to either configure a service or construct an authenticator.

Reply:
I copied this logic from above so I didn't look too much into but I agree with your suggestion. I made it so only the well-defined properties are returned